### PR TITLE
raftstore-v2: fix issues related to background work

### LINKED ISSUE #15145: [Dynamic Region] QPS continues to decrease after inject one tikv network partition for 50m and recover due to flow controller
### What version of TiKV are you using?
./tikv-server -V
 TiKV 
Release Version:   7.3.0-alpha
Edition:           Community
Git Commit Hash:   31a629950d05f893fdd72717938118ad11ed5d3f
Git Commit Branch: heads/refs/tags/v7.3.0-alpha
UTC Build Time:    2023-07-15 11:12:10
Rust Version:      rustc 1.67.0-nightly (96ddd32c4 2022-11-14)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release
2023-07-17T15:20:50.746+0800

### What operating system and CPU are you using?
8c/32g

### Steps to reproduce
1、run tpcc with 20000 warehouse and 32 threads 
2、inject one tikv network partition for 50m
inject fault：2023/07/17 15:30:56.693 +08:00
recover fault：2023/07/17 16:20:56.760 +08:00

### What did you expect?
qps can recover to n-1/n after injection fault and can recover after  fault recover

### What did happened?
 QPS continues to decrease after inject and recover fault
image
img_v2_d480c9ed-9c92-48b1-8985-b707e2ce86bg
img_v2_da07605a-9de4-4970-a097-49db9dd7a97g

### PR DESCRIPTION
**Pull request by tabokie**
### What is changed and how it works?

Issue Number: Ref #12842 Fix #15145

What's Changed:

```commit-message
- Raise error when block-cache or write-buffer-limit is set too high.
- Reduce write-buffer-limit ratio and max size.
- Fix `max-compactions` adjusting doesn't consider level zero files.
- Increase `max-manual-flush-rate` from 0.2 to 0.3.
- Fix pre flush doesn't call callback when tablet is missing.
- Move `pause_background_work` into background pool because it may block for a long time.
- Only split clean tablet.
- Schedule pre flush for commit merge in a dedicated merge because it blocks the source region.
- Print rocksdb stats once 2h for v2.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by tonyxuqqi — APPROVED** [review]

File: components/server/src/common.rs [hunk 1]
```diff
@@ -514,17 +516,34 @@ impl EnginesResourceInfo {
...
                 }
+                if let Ok(Some(n)) = engine.get_cf_num_files_at_level(cf, 0) {
+                    let level0 = n as f32;
+                    let slowdown_trigger = cf_opts.get_level_zero_slowdown_writes_trigger() as f32;
+                    let compaction_trigger =
+                        cf_opts.get_level_zero_file_num_compaction_trigger() as f32;
+                    let ratio = if slowdown_trigger > compaction_trigger {
+                        (level0 - compaction_trigger) / (slowdown_trigger - compaction_trigger)
+                    } else {
+                        1.0
```
Comment (SpadeA-Tang) [inline components/server/src/common.rs:540]: Is it mean if slowdown_trigger <= compaction_trigger, we disable the limiter?

File: src/storage/config.rs [hunk 2]
```diff
@@ -276,21 +277,11 @@ impl BlockCacheConfig {
...
-                let total_mem = SysQuota::memory_limit_in_bytes();
-                if engine_type == EngineType::RaftKv2 {
-                    ((total_mem as f64) * RAFTSTORE_V2_BLOCK_CACHE_RATE) as usize
-                } else {
-                    ((total_mem as f64) * BLOCK_CACHE_RATE) as usize
-                }
-            }
-            Some(c) => c.0 as usize,
-        };
+        let capacity = self.capacity.unwrap_or(FALLBACK_BLOCK_CACHE_CAPACITY).0 as usize;
```
Comment (SpadeA-Tang) [inline src/storage/config.rs:284]: It means we have to set the cache size manually?

Comment (tabokie) [reply src/storage/config.rs:284]: Yes

File: components/server/src/common.rs (hunk 1 above)
Comment (tabokie) [reply components/server/src/common.rs:540]: Yes.

File: src/storage/config.rs (hunk 2 above)
Comment (tabokie) [reply src/storage/config.rs:284]: See the comment for `FALLBACK_BLOCK_CACHE_CAPACITY`. It's set in validate.

**Review by SpadeA-Tang — APPROVED** [review]

**Pull request merged**